pyparsing/core.py

```diff
@@ -297,21 +297,20 @@ def wrapper(*args):
                 # re-raise TypeErrors if they did not come from our arity testing
                 if found_arity:
                     raise
-                else:
```

Comment:
Please remove all the changes to core.py from this PR. They are getting confused with the other.